docs/FAQs.md

```diff
@@ -13,6 +13,131 @@ After you set up a repository, you can substitute in any tools you'd like.
 
+## Can I create a GitHub action with CTA?
+
+Yes! If you want to read in detail about GitHub Actions [try this](https://docs.github.com/en/actions/creating-actions).
+Here we'll outline the steps required to migrate a CTA app to a GitHub Action:
+
+1. The GitHub Actions does not cater for tabs well, so you will likely want to set your `README.md` to use spaces, not tabs.
+2. GitHub Actions are _not_ like packages published to npm.
+   The output is expected to live in the `dist` folder of a GitHub repo.
+   As a consequence we should:
+
+   - delete `.github/workflows/release.yml` and `.github/workflows/post-release.yml`.
+   - update `.github/workflows/build.yml`
+
+   ```diff
+   -      - run: node ./lib/index.js
+   ```
+
+   - GitHub Actions have a different build mechanism.
+     So you'll be removing `tsup` in favour of [`ncc`](https://github.com/vercel/ncc) which compiles output into a single JS file.
+     Delete `tsup.config.ts` then execute the following commands:
+
+   ```bash
+   pnpm remove tsup
+   pnpm add @vercel/ncc -D
+   pnpm add @actions/core -P
+   ```
+
+   - We also added the [`@actions/core`](https://github.com/actions/toolkit/tree/master/packages/core) package, used for building GitHub Actions.
+     Now we need to update the `build` script in our `package.json`:
+
+   ```diff
+   -"build": "tsup",
+   +"build": "ncc build src/index.ts -o dist --license licenses.txt",
+   ```
+
+   - Our build now emits to the `dist` directory; so we'll want to avoid linting that directory by adding the following to `.eslintignore` and our `.prettierignore`:
+
+   ```diff
+   +dist
+   ```
+
+   - Rather than having to remember to compile each time, we'll update our precommit hook in `.husky/precommit` to build for us on each commit:
+
+   ```diff
+   +pnpm run build
+   +git add dist
+   npx lint-staged
+   ```
+
+   - on the off chance someone isn't running husky, we'll add `.github/workflows/check-dist.yml` ([based on this](https://github.com/actions/typescript-action/blob/main/.github/workflows/check-dist.yml)) to ensure `dist` is up to date:
+
+   ```yml
+   # In TypeScript actions, `dist/` is a special directory. When you reference
+   # an action with the `uses:` property, `dist/index.js` is the code that will be
+   # run. For this project, the `dist/index.js` file is transpiled from other
+   # source files. This workflow ensures the `dist/index.js` file contains the
+   # expected transpiled code.
+   #
+   # If this workflow is run from a feature branch, it will act as an additional CI
+   # check and fail if the checked-in `dist/` directory does not match what is
+   # expected from the build.
+   jobs:
+     check-dist:
```

Comment:
[Style] Other workflows have kept to minimal names like `lint`. Any thoughts on renaming this to `dist.yml` + `jobs.dist`?

The minimal-ness isn't _quite_ consistent given names like `lint-knip` and `lint-spelling`... but I think once those get consolidated into ESLint (https://github.com/JoshuaKGoldberg/create-typescript-app/milestone/3: #567, #839, #897) we'll be in a good place to rename scripts and their workflows to `knip`, `lint`, `packages`, etc.

Not a blocker at all. 😄

Reply:
Yeah I guess dist.yml would be fine... An alternative idea would be to combine the dist check with the `build.yml` - thoughts?

Reply:
Oh I like that!

Which reminds me - I think there should be _two_ build task:

* Release builds: which is `pnpm build:release` in my repos
* Type checking: which is `pnpm build` in my repos

Reply:
I'll move the `check-dist` stuff into `build` shortly.

> Release builds: which is pnpm build:release in my repos
> Type checking: which is pnpm build in my repos

I find this naming slightly confusing. The question it prompts to me is "I have `build` and `build:release` - are they building different things? Does it matter?"  If it was just `release` and not `build:release` I think I'd think "okay, that one's pushing out to npm".  But `build:release` makes me do a double take

Reply:
Hmm yeah. Maybe it should be `"build" "ncc ..."` and `"tsc": "tsc"`? Thoughts?

Reply:
I actually don't love `tsc` as a name. Questions it prompts: "tsc can output js and typecheck, I wonder what tsc means here?" Ideally names (for my money) should prompt as few questions as possible. How about `typecheck`? (As I think tsc is just being used for type checking?)

Reply:
Blurgh there's something about `typecheck` that saddens me as a name... All the other ones that don't have a `:` are one word. Then this is two. And so long!

Agreed with the gripes about `tsc`. I don't love it either. 😦